Bankersi2k24-web/src/main/java/com/isa/Bankersi2k24/services/UserService.java

```diff
@@ -83,7 +83,15 @@ public String getUserName(BigInteger id) throws Exception {
             throw new Exception(String.format("User with id: %d does not exist", id));
         }
     }
-
+    public User getUserById(BigInteger id){
+        try{
+            return this.fetchAllUsers().stream().filter(u -> u.getId().equals(id))
+                    .findFirst().get();
+        }catch (Exception e){
+            e.printStackTrace();
+            return null;
+        }
+    }
```

Comment:
```suggestion
    public User getUserById(BigInteger id){
        return userRepository.findById(id).orElseThrow(
        () -> new EntityNotFoundException(String.format("Not found user with given Id:  %s", id));
    }
```
odwołałeś się do fetchAllUser, które w lini 73 zwraca null w wyniku czego zawsze otrzymasz nullPointerException. 
Użyłem userRepository z interface jpaRepository, któy ma przygotowaną definicję findById();